addons/purchase/tests/test_purchase.py

```diff
@@ -493,6 +493,14 @@ def test_purchase_multicurrency(self):
         self.assertEqual(order_b.order_line.price_unit, 10.0, 'The price unit should be 10.0')
         self.assertEqual(order_b.amount_total_cc, order_b.amount_total, 'Company Total should be 10.0$')
 
+        with self.assertRaises(KeyError):
+            order_b.tax_totals['amount_total_cc']
+        order_b.currency_id = company_a.currency_id
+        self.assertEqual(
+            order_b.tax_totals['total_amount'] / order_b.tax_totals['total_amount_currency'],
+            company_b.currency_id['rate']/company_a.currency_id['rate'],
+        )
```

Comment:
The test passes even without your change, so it doesn't seem to check that the error you're trying to fix doesn't come back in the future :confused: 
And as the codes only adds a `amount_total_cc` to the `tax_totals`, I don't think this assert is relevant to what you want to check.